extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/GreenbidsRealTimeDataProcessedAuctionRequestHook.java

```diff
@@ -0,0 +1,488 @@
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.hooks.execution.v1.auction.AuctionRequestPayloadImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.Partner;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.ThrottlingThresholds;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.AnalyticsResult;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.ExplorationResult;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.GreenbidsUserAgent;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.ModelCache;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.OnnxModelRunner;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.Ortb2ImpExtResult;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.ThresholdCache;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.ThrottlingMessage;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.InvocationResultImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.ActivityImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.AppliedToImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.ResultImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.TagsImpl;
+import org.prebid.server.hooks.v1.InvocationAction;
+import org.prebid.server.hooks.v1.InvocationResult;
+import org.prebid.server.hooks.v1.InvocationStatus;
+import org.prebid.server.hooks.v1.analytics.Result;
+import org.prebid.server.hooks.v1.analytics.Tags;
+import org.prebid.server.hooks.v1.auction.AuctionInvocationContext;
+import org.prebid.server.hooks.v1.auction.AuctionRequestPayload;
+import org.prebid.server.hooks.v1.auction.ProcessedAuctionRequestHook;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.StreamSupport;
+
+public class GreenbidsRealTimeDataProcessedAuctionRequestHook implements ProcessedAuctionRequestHook {
```

Comment:
the hook class looks overwhelmed, ideally I'd try to shrink the hook code to:
1. calling a filter and getting filtering result
2. prepare invocation result

everything else is better to be extracted to separate services

Reply:
Extracted the logic into 4 classes
- `GreenbidsInferenceData`
- `OnnxModelRunnerWithThresholds`
- `FilterService`
- `GreenbidsInvocationResult`